dart-mode.el

```diff
@@ -839,6 +862,40 @@ true for positions before the start of the statement, but on its line."
          ((?{) (dart-in-block-p (c-guess-basic-syntax))))))))
 
 
+(defun dart--process-nav-info (response offset)
+  "Report the possible completions and jump to the location of the definition.
+
+  Opens a new file in a new buffer if necessary."
+  (dart-info (format "Reporting navigation : %s" response))
+  (-when-let* ((regions (cdr (assq 'regions (assq 'result response))))
+	       (target-offset
+		(loop for region across regions
+		      if (equal offset (cdr (assoc 'offset region)))
+		      return (aref (cdr (assoc 'targets region)) 0)))
+	       (target (aref (cdr (assoc 'targets (assoc 'result response)))
```

Comment:
Rather than writing `(assoc 'result response)` over and over, just assign it to a variable earlier in the let.